test/tests/ver-1kd/ver-1kd.i

```diff
@@ -0,0 +1,637 @@
+unit_scale = 1
+unit_scale_neighbor = 1
+
+[Mesh]
+  [generated]
+    type = GeneratedMeshGenerator
+    dim = 1
+    nx = ${nb_segments_TMAP8}
+    xmax = ${long_total}
+  []
+  [enclosure_1]
+    type = SubdomainBoundingBoxGenerator
+    input = generated
+    block_id = 1
+    bottom_left = '0 0 0'
+    top_right = '${fparse 1/3 * long_total} 0 0'
+  []
+  [enclosure_2]
+    type = SubdomainBoundingBoxGenerator
+    input = enclosure_1
+    block_id = 2
+    bottom_left = '${fparse 1/3 * long_total} 0 0'
+    top_right = '${fparse long_total} 0 0'
+  []
+  [interface]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = enclosure_2
+    primary_block = 1
+    paired_block = 2
+    new_boundary = interface
+  []
+  [interface2]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = interface
+    primary_block = 2
+    paired_block = 1
+    new_boundary = interface2
+  []
+[]
+
+[Variables]
+
+  # Variables for H2, T2, and HT in enclosure 1
+  [concentration_H2_enclosure_1]
+    block = 1
+    initial_condition = '${initial_concentration_1}'
+  []
+  [concentration_T2_enclosure_1]
+    block = 1
+    initial_condition = '${initial_concentration_1}'
+  []
+  [concentration_HT_enclosure_1]
+    block = 1
+    initial_condition = '${initial_concentration_2}'
+  []
+
+  # Variables for H2, T2, and HT in enclosure 2
+  [concentration_H2_enclosure_2]
+    block = 2
+    initial_condition = '${initial_concentration_2}'
+  []
+  [concentration_T2_enclosure_2]
+    block = 2
+    initial_condition = '${initial_concentration_2}'
+  []
+  [concentration_HT_enclosure_2]
+    block = 2
+    initial_condition = '${initial_concentration_2}'
+  []
+[]
+
+[AuxVariables]
+
+  # Auxiliary variables for H2, T2, and HT in enclosure 1
+  [pressure_H2_enclosure_1]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 1
+    initial_condition = '${initial_pressure_1}'
+  []
+  [pressure_T2_enclosure_1]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 1
+    initial_condition = '${initial_pressure_1}'
+  []
+  [pressure_HT_enclosure_1]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 1
+    initial_condition = '${initial_pressure_2}'
+  []
+
+  # Auxiliary variables for H2, T2, and HT in enclosure 2
+  [pressure_H2_enclosure_2]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 2
+    initial_condition = '${initial_pressure_2}'
+  []
+  [pressure_T2_enclosure_2]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 2
+    initial_condition = '${initial_pressure_2}'
+  []
+  [pressure_HT_enclosure_2]
+    order = CONSTANT
+    family = MONOMIAL
+    block = 2
+    initial_condition = '${initial_pressure_2}'
+  []
+[]
+
+[Kernels]
+  # Source term for T2 in enclosure 1
+  [T2_source_term_enclosure_1]
+    type = BodyForce
+    variable = concentration_T2_enclosure_1
+    block = '1'
+    value = '${fparse source/volume * kb * temperature}'
+    function = 'x+y'
```

Comment:
```suggestion
```
why were you using this? this would make the body force equal to 
`S*(x+y)`, which you do not want to do here.